File: pydss/general/TSDataObjImpl.py
from dss.general.TSDataObj import (TSDataObj,)
from dss.general.impl.CableDataObjImpl import (CableDataObjImpl,)
from dss.general.TSData import (TSData,)


class TSDataObjImpl(CableDataObjImpl, TSDataObj):
    diaShield = None
    tapeLayer = None
    tapeLap = None

    def __init__(self, parClass, TSDataName):
        super(TSDataObjImpl, self)(parClass, TSDataName)
        self.setName(TSDataName.toLowerCase())
        self.objType = parClass.getDSSClassType()
        self.diaShield = -1.0
        self.tapeLayer = -1.0
        self.tapeLap = 20.0
        self.initPropertyValues(0)

    def dumpProperties(self, f, complete):
        super(TSDataObjImpl, self).dumpProperties(f, complete)
        _0 = True
        i = 0
        while True:
            if _0 is True:
                _0 = False
            else:
                i += 1
            if not (i < self.getParentClass().getNumProperties()):
                break
            f.print_('~ ' + self.getParentClass().getPropertyName()[i] + '=')
            _1 = i
            _2 = False
            while True:
                if _1 == 0:
                    _2 = True
                    print String.format.format('%.6g', self.diaShield)
                    break
                if (_2 is True) or (_1 == 1):
                    _2 = True
                    print String.format.format('%.6g', self.tapeLayer)
                    break
                if (_2 is True) or (_1 == 2):
                    _2 = True
                    print String.format.format('%.2g', self.tapeLap)
                    break
                break

    def initPropertyValues(self, arrayOffset):
        self.setPropertyValue(0, '-1')
        self.setPropertyValue(1, '-1')
        self.setPropertyValue(2, '20.0')
        super(TSDataObjImpl, self).initPropertyValues(arrayOffset + TSData.NumPropsThisClass)

    def getDiaShield(self):
        return self.diaShield

    def getTapeLayer(self):
        return self.tapeLayer

    def getTapeLap(self):
        # FIXME Private members in OpenDSS
        return self.tapeLap

    def setDiaShield(self, shield):
        self.diaShield = shield

    def setTapeLayer(self, layer):
        self.tapeLayer = layer

    def setTapeLap(self, lap):
        self.tapeLap = lap

File: core/dss-core/src/main/java/com/ncond/dss/general/impl/TSDataObjImpl.java
package com.ncond.dss.general.impl;

import java.io.OutputStream;
import java.io.PrintWriter;

import com.ncond.dss.common.DSSClass;
import com.ncond.dss.general.TSData;
import com.ncond.dss.general.TSDataObj;

public class TSDataObjImpl extends CableDataObjImpl implements TSDataObj {

	private double diaShield;
	private double tapeLayer;
	private double tapeLap;

	public TSDataObjImpl(DSSClass parClass, String TSDataName) {
		super(parClass, TSDataName);

		setName(TSDataName.toLowerCase());
		objType = parClass.getDSSClassType();
		diaShield = -1.0;
		tapeLayer = -1.0;
		tapeLap = 20.0;

		initPropertyValues(0);
	}

	@Override
	public void dumpProperties(OutputStream out, boolean complete) {
		super.dumpProperties(out, complete);

		PrintWriter pw = new PrintWriter(out);

		for (int i = 0; i < getParentClass().getNumProperties(); i++) {
			pw.print("~ " + getParentClass().getPropertyName(i) + "=");
			switch (i) {
			case 0:
				pw.printf("%.6g", diaShield);
				break;
			case 1:
				pw.printf("%.6g", tapeLayer);
				break;
			case 2:
				pw.printf("%.2g", tapeLap);
				break;
			}
			pw.println();
		}
	}

	@Override
	public void initPropertyValues(int arrayOffset) {
		setPropertyValue(0, "-1");
		setPropertyValue(1, "-1");
		setPropertyValue(2, "20.0");
		super.initPropertyValues(arrayOffset + TSData.NumPropsThisClass);
	}

	@Override
	public double getDiaShield() {
		return diaShield;
	}

	@Override
	public double getTapeLayer() {
		return tapeLayer;
	}

	@Override
	public double getTapeLap() {
		return tapeLap;
	}

	// FIXME Private members in OpenDSS

	@Override
	public void setDiaShield(double shield) {
		diaShield = shield;
	}

	@Override
	public void setTapeLayer(double layer) {
		tapeLayer = layer;
	}

	@Override
	public void setTapeLap(double lap) {
		tapeLap = lap;
	}

}
